temporal-sdk/src/main/java/io/temporal/common/converter/DataConverter.java

```diff
@@ -108,6 +109,24 @@ <T> T fromPayloads(
       int index, Optional<Payloads> content, Class<T> valueType, Type valueGenericType)
       throws DataConverterException;
 
+  /**
+   * Instantiate an appropriate Java Exception from a serialized Failure object. The default
+   * implementation delegates the conversion process to an instance of {@link
+   * io.temporal.failure.FailureConverter}, using this data converter for payload decoding.
+   *
+   * @param failure Failure protobuf object to deserialize into an exception
+   */
+  RuntimeException failureToException(Failure failure);
```

Comment:
I saw a null check later. Do you need to add nullability annotations?

Reply:
I would suggest marking this method non-null, throw on failure=null and add a javadoc about it. 

I even may see some value for exceptionToFailure to support null. While it shouldn't, but we have situations where we explicitly send encoded "null" Payload, for example as a workflow or activity result.

But there is no reason for a decoding method to support null and I think it will make the contract of this method unclear. Callers of this method should be dealing with null and avoid calling it.

Reply:
After verification, `exceptionToFailure` is never called with a null exception. I added `@Nonnull` on that one too.